packages/core/src/browser/tree/tree-expansion.spec.ts

```diff
@@ -0,0 +1,158 @@
+import { TreeSelectionService } from './tree-selection';
+import { TreeExpansionServiceImpl, TreeExpansionService, ExpandableTreeNode } from './tree-expansion';
+import { TreeNavigationService } from './tree-navigation';
+import { TreeSearch } from './tree-search';
+import { FuzzySearch } from './fuzzy-search';
+import { MockLogger } from '../../common/test/mock-logger';
+import { ILogger } from '../../common';
+
+// tslint:disable:no-unused-expression
+describe('TreeExpansionService', () => {
+  let model: TreeModel;
+  beforeEach(() => {
+    model = createTreeModel();
+    model.root = MockTreeModel.HIERARCHICAL_MOCK_ROOT();
+  });
+  describe('expandNode', () => {
+    it('won\'t expand an already expanded node', done => {
+      const node: ExpandableTreeNode = retrieveNode<ExpandableTreeNode>('1');
+      model.expandNode(node).then(result => {
+        expect(result).to.be.false;
+        done();
+      });
+    });
+
+    it('will expand a collapsed node', done => {
+      const node: ExpandableTreeNode = retrieveNode<ExpandableTreeNode>('1');
+      model.collapseNode(node).then(() => {
+        model.expandNode(node).then(result => {
+          expect(result).to.be.true;
+          done();
+        });
+      });
+    });
+
+    it('won\'t expand an undefined node', done => {
+      model.expandNode(undefined).then(result => {
+        expect(result).to.be.false;
+        done();
+      });
+    });
+  });
+
+  describe('collapseNode', () => {
+    it('will collapse an expanded node', done => {
+      const node: ExpandableTreeNode = retrieveNode<ExpandableTreeNode>('1');
+      model.collapseNode(node).then(result => {
+        expect(result).to.be.true;
+        done();
+      });
+    });
+
+    it('won\'t collapse an already collapsed node', done => {
+      const node: ExpandableTreeNode = retrieveNode<ExpandableTreeNode>('1');
+      model.collapseNode(node).then(() => {
+        model.collapseNode(node).then(result => {
+          expect(result).to.be.false;
+          done();
+        });
+      });
+    });
+
+    it('cannot collapse a leaf node', done => {
+      const node: ExpandableTreeNode = retrieveNode<ExpandableTreeNode>('1.1.2');
+      model.collapseNode(node).then(result => {
+        expect(result).to.be.false;
+        done();
+      });
+    });
+  });
+
+  describe('collapseAll', () => {
+    it('will collapse all nodes recursively', done => {
+      model.collapseAll(retrieveNode<CompositeTreeNode>('1')).then(result => {
+        expect(result).to.be.true;
+        done();
+      });
+    });
+
+    it('won\'t collapse nodes recursively if the root node is collapsed already', done => {
+      model.collapseNode(retrieveNode<ExpandableTreeNode>('1')).then(() => {
+        model.collapseAll(retrieveNode<CompositeTreeNode>('1')).then(result => {
+          expect(result).to.be.true;
+          done();
+        });
+      });
+    });
+  });
+
+  describe('toogleNodeExpansion', () => {
```

Comment:
minor typo, `toogle` should be `toggle`.

Reply:
should be fixed with 3bfb5dfa44eacabcbb17a64aa0544ffe122b5fdf